app/services/api/charts/cp_benchmark.rb

```diff
@@ -0,0 +1,117 @@
+module Api
+  module Charts
+    class CPBenchmark
+      # Calculate companies stats grouped by CP benchmark in multiple series.
+      # @return [Array<Hash> chart data
+      # @example
+      #   [
+      #     {
+      #       name: 'Below 2',
+      #       data: [ ['Coal Mining', 52], ['Steel', 73] ]
+      #     },
+      #     {
+      #       name: 'Paris',
+      #       data: [ ['Coal Mining', 65], ['Steel', 26] ]
+      #     }
+      #   ]
+      def cp_performance
+        prepare_chart_data(sectors_cp_performance)
+      end
+
+      private
+
+      # Prepare data for chart based on sectors cp performance data
+      # @param results[Array<Hash>] list with all scenarios with emissions
+      # @return [Array<Hash>] list with data for chart
+      def prepare_chart_data(results)
+        results.map do |scenario, companies|
+          data = companies.map do |company, value|
+            [company, value]
+          end
+
+          {name: scenario, data: data}
+        end
+      end
+
+      # Scenarios with sectors and companies count
+      # @return [Hash]
+      # @example
+      #   {
+      #     "2 Degrees (High Efficiency)" => {
+      #       "Aluminium" => 10,
+      #       "Cement" => 27
+      #     },
+      #       "International Pledges" => {
+      #       "Steel" => 80
+      #     }
+      #   }
+      def sectors_cp_performance
+        results = {}
+
+        sectors_scenario_emissions.each do |sector, scenarios|
+          sector.companies.each do |company|
+            next if scenarios.empty?
+
+            company_emission = company_emission(company)
+
+            next unless company_emission
+
+            company_scenario = company_scenario(company_emission, scenarios)
+
+            results[company_scenario] ||= Hash.new(0)
```

Comment:
My main concern here is that because we start building this data from **sectors**, we have to perform this hash-based aggregation/counting in 2 places.

Please check https://github.com/Vizzuality/laws_and_pathways/pull/69 - although it still doesn't work, maybe we can give it a try?:)

Reply:
@kowal it start building from `sectors`. `company_scenario` here means which sector scenario match to current emission for company.
I'll take a look on your pull request :)